app/Repositories/Membership/MembershipRepository.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Repositories\Membership;
+
+use App\User;
+use App\ActivityMetric;
+
+class MembershipRepository
```

Comment:
We extend Repository from BaseRepository and implement Interface in current practice. Please check how Project/Activity Repositories has been implemented

Reply:
Yes, I understand. But the Metrics and Membership repositories don't really represent a particular model so implementing the EloquentRepositoryInterface wont be useful for them.

They are both calculations based on different models; not meant to perform CRUD operations nor query most of the calculated fields. Most of the interface implementations would have to remain empty.

Do you still want me to extend baserepo just for style?

Reply:
If we do not have model then it make sense to me.